bundles/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/gson/KonnectedModuleGson.java

```diff
@@ -36,12 +41,50 @@
     private Integer pollInterval;
     private String addr;
 
-    public Integer getPin() {
-        return pin;
+    private Integer getPin() {
+
+        return Arrays.asList(PIN_TO_ZONE).indexOf(pin);
     }
 
-    public void setPin(Integer setPin) {
-        this.pin = setPin;
+    private void setPin(Integer setPin) {
+
+        this.pin = Arrays.asList(PIN_TO_ZONE).get(setPin);
```

Comment:
Are you aware that your code crashes if `setPin` has a value that doesn't exist in the array?